dotnet/src/SemanticKernel/TemplateEngine/PromptTemplateEngine.cs

```diff
@@ -85,6 +93,12 @@ public async Task<string> RenderAsync(IList<Block> blocks, SKContext context)
             }
         }
 
+        var codeBlocks = await Task.WhenAll(tasks).ConfigureAwait(false);
+        for (var i = 0; i < guids.Count; i++)
+        {
+            result.Replace(guids[i].ToString(), codeBlocks[i]);
```

Comment:
foreach (Task<string> t in 
```suggestion
        foreach (Task<string> t in tasks)
        {
            result.Append(await t.ConfigureAwait(false));
        }
```
This allows the tasks to all run in parallel, but they'll be processed in order. Static blocks are instantly completed (awaiting a Task.FromResult is a noop). You don't need to WaitAll to get started with the string builder... just need to await each task in order to ensure it's completed by the time you get to it.

Reply:
thanks for this suggestion, added